webapp/src/js/controllers/reports.js

```diff
@@ -49,6 +51,34 @@ angular
 
     var lineage = lineageFactory();
 
+    // Render the facilities hierarchy as the user is scrolling through the list
+    // Initially, don't load/render any
+    $scope.totalFacilitiesDisplayed = 0;
+    $scope.facilities = [];
+
+    // Load the facilities hierarchy and render one district hospital 
+    // when the user clicks on the filter dropdown
+    $scope.monitorFacilityDropdown = () => {
+      PlaceHierarchy()
+        .then(function(hierarchy) {
+          $scope.facilities = hierarchy;
+          $scope.totalFacilitiesDisplayed += 1;
```

Comment:
The first page should be at least 20 rows otherwise it'll have to go through the scroll event listener multiple times until the scrollbar appears.

Reply:
Similar to previous comment .. in this case, the user gets the loading spinner and will not immediately get something if it takes time to render the top 20 facilities and their children.

Reply:
Makes sense.